File: briar-android/src/main/java/org/briarproject/briar/android/contact/BriarInboxThreadItem.java
package org.briarproject.briar.android.contact;

import org.briarproject.nullsafety.NotNullByDefault;

@NotNullByDefault
final class BriarInboxThreadItem implements InboxThreadItem {

	private final ContactListItem item;

	BriarInboxThreadItem(ContactListItem item) {
		this.item = item;
	}

	ContactListItem getItem() {
		return item;
	}

	@Override
	public String getStableId() {
		return "briar:" + item.getContact().getId().getInt();
	}

	@Override
	public long getLatestActivityMillis() {
		return item.getTimestamp();
	}

	@Override
	public Source getSource() {
		return Source.BRIAR;
	}
}

File: briar-android/src/main/kotlin/org/briarproject/briar/android/contact/BriarInboxThreadItem.kt
package org.briarproject.briar.android.contact

import org.briarproject.briar.api.connector.ConnectorSource

class BriarInboxThreadItem(val item: ContactListItem) : InboxThreadItem {

	override val stableId: String
		get() = "briar:${item.contact.id.int}"

	override val latestActivityMillis: Long
		get() = item.timestamp

	override val connectorSource: ConnectorSource?
		get() = null
}
